include/revng/Support/GraphAlgorithms.h

```diff
@@ -400,7 +418,11 @@ class DFSReachableState : public DFState<NodeT> {
     // Check that, if we are trying to insert a block which is the `Targets`
     // set, we add all the nodes on the current visiting stack in the
     // `Targets` set.
-    if (Targets.contains(Block)) {
+    auto BeginIt = llvm::GraphTraits<NodeT>::child_begin(Block);
+    auto EndIt = llvm::GraphTraits<NodeT>::child_end(Block);
+    size_t BlockSuccessorsN = std::distance(BeginIt, EndIt);
+    if ((Targets.contains(Block))
+        or ((Target == nullptr) && (BlockSuccessorsN == 0))) {
```

Comment:
Drop extra parentheses around the whole `if` condition.

Reply:
Done, this part of the code is now in `revng-c`.